app.py

```diff
@@ -114,40 +114,64 @@ def handle_put_experience():
 
     return jsonify({"Server Error": "Couldn't process method"})
 
-
-@app.route('/resume/education', methods=['GET', 'POST'])
+@app.route('/resume/education', methods=['GET', 'POST', 'DELETE'])
 def education():
     '''
     Handles education requests
     '''
     if request.method == 'GET':
-        index = request.args.get("index")
-        if index is not None:
-            return get_education_by_index(data, index)
-        return jsonify(data["education"])
+        return handle_get_education()
 
     if request.method == 'POST':
-        req = request.get_json()
+        return handle_post_education()
 
-        required_fields = {"school":"string", "start_date":"string", "end_date":"string" \
-                           , "grade":"string", "logo":"string"}
+    if request.method == 'DELETE':
+        return handle_delete_education()
 
-        code, err_message = validate_request(req, required_fields)
+    return jsonify({"Server Error": "Couldn't process method"})
 
-        if code != 0:
-            return jsonify({"error": err_message}), code
+def handle_get_education():
+    '''
+    Handle education get requests
+    '''
+    index = request.args.get("index")
+    if index is not None:
+        return get_education_by_index(data, index)
+    return jsonify(data["education"])
 
-        new = Education(req["course"],
-            req["school"],
-            req["start_date"],
-            req["end_date"],
-            req["grade"],
-            req["logo"]
-        )
-        data["education"].append(new)
+def handle_post_education():
+    '''
+    Handle education post requests
+    '''
+    req = request.get_json()
+
+    required_fields = {"school":"string", "start_date":"string", "end_date":"string" \
+                        , "grade":"string", "logo":"string"}
+
+    code, err_message = validate_request(req, required_fields)
+
+    if code != 0:
+        return jsonify({"error": err_message}), code
+
+    new = Education(req["course"],
+        req["school"],
+        req["start_date"],
+        req["end_date"],
+        req["grade"],
+        req["logo"]
+    )
+    data["education"].append(new)
+    return jsonify({"id": data["education"].index(new)})
+
+def handle_delete_education():
+    '''
+    Handle education delete requests
+    '''
+    index = request.args.get("index")
+    if index is not None:
+        return delete_education_by_index(data, index)
+    return jsonify(data["education"])
```

Comment:
This isn't perhaps the clearest. If they don't include an index, it just returns all the education. I would think about returning some sort of error message @ganavijayaram

Reply:
Have made the necessary changes.